warehouse/core/src/main/java/datawave/ingest/mapreduce/job/SplittableRFileInputFormat.java

```diff
@@ -0,0 +1,137 @@
+import org.apache.accumulo.core.file.FileSKVIterator;
+import org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile;
+import org.apache.accumulo.core.file.rfile.RFile;
+import org.apache.accumulo.core.spi.crypto.CryptoEnvironment;
+import org.apache.accumulo.core.spi.crypto.CryptoService;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.apache.log4j.Logger;
+
+public class SplittableRFileInputFormat extends RFileInputFormat {
+    private static final Logger log = Logger.getLogger(SplittableRFileInputFormat.class);
+    public static final String MIN_BLOCKS_PER_SPLIT = SplittableRFileInputFormat.class.getName() + ".minBlocksPerSplit";
+
+    @Override
+    protected boolean isSplitable(JobContext context, Path filename) {
+        return true;
+    }
+
+    @Override
+    public RecordReader<Key,Value> createRecordReader(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+        return new SplittableRFileRecordReader();
+    }
+
+    /**
+     * Create rfile splits for the configured job. See <code>FileInputFormat</code>
+     *
+     * @param job
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public List<InputSplit> getSplits(JobContext job) throws IOException {
+        Configuration config = job.getConfiguration();
+
+        List<InputSplit> rfileSplits = new ArrayList<>();
+        log.info("getting splits for job");
+        // get the configured directories/files
+        for (InputSplit inputSplit : super.getSplits(job)) {
+            if (!(inputSplit instanceof FileSplit)) {
+                throw new IllegalArgumentException("Must have file splits");
+            }
+
+            // for each file get the index blocks
+            FileSplit fileSplit = (FileSplit) inputSplit;
+            log.info("creating splits for: " + fileSplit);
+            rfileSplits.addAll(getSplits(config, fileSplit));
+        }
+        log.info("total splits: " + rfileSplits.size());
+
+        return rfileSplits;
+    }
+
+    /**
+     * Create rfile splits for a given FileSplit. Splits will honor the MIN_BLOCKS_PER_SPLIT config option. Splits may extend more than MIN_BLOCKS_PER_SPLIT if
+     * a given Key spans more index blocks.
+     *
+     * @param config
+     * @param fileSplit
+     * @return
+     * @throws IOException
+     */
+    public static List<InputSplit> getSplits(Configuration config, FileSplit fileSplit) throws IOException {
+        log.info("getting splits for: " + fileSplit);
+        Path rfile = fileSplit.getPath();
+        FileSystem fs = rfile.getFileSystem(config);
+
+        if (!fs.exists(rfile)) {
```

Comment:
Do you see us hitting this condition? It looks like we'll only get a `FileSplit` from `FileInputFormat` if it exists. This would be like an early catch if the RFile got compacted I guess? Just wondering if we should forego this for the sake of saving the roundtrips to the namenode when this would only trip in a narrow window between `FileInputFormat.getSplits()` and this call.

Reply:
The check really isn't there for that case at all, but instead because the method is public, when being called from getSplits(JobContext) I don't believe this is a case to worry about.